### DISCUSSION TITLE
Niko HC polling in HA (delay in state - so delay in automation)

### DISCUSSION THREAD

**Discussion by BartSymfonielaan** [joleys/niko-home-control-II#199] [Q&A] [answered]
This integration is working great, however i learned HA only polls states every 1 minute.

So if i make an automation (put wifi plug on when i do this) there is a significant delay if not even carried out.

Can you make this faster and if so where stated in the config file?

Thanks!

**Comment by tijsverkoyen** [comment]
This is not how the integration works.
The integration is "realtime", as in the events are processed as they are received.

I don't know which devices you are using, or what your automation exactly does. But to my knowledge Niko does not have Wifi plugs, the plugs are Zigbee.

**Comment by BartSymfonielaan** [comment]
Hi Tijs

Thanks for the response. I belief Niko polls states at certain intervals not push? Anyways if i open my blindings of the house then my electric curtains must open - but here sometimes it takes more than a minute to exectute this command. But also on the wifi plugs (these are Tuya attached) for example.

It also sometimes misses such an event not triggering the needed result.

**Comment by tijsverkoyen** [comment, marked as answer]
> I belief Niko polls states at certain intervals not push?

No. Internally the Niko Home Controller API is a MQTT based. So as soon as the event is pushed by the Niko Home Controller on MQTT the integration processes it. So it is "push".

You can enable debugging for this integration and will see all the events in the log with the timestamp. If the "event" is received with a delay I suggest you contact Niko support.

My best guess is that the "event" is triggered in the Niko Home Controller only when the blinds are fully open. But I don't over any blinds, so I can't verify that. But you can check the logs.
